lib/render/game.php

```diff
@@ -288,3 +290,31 @@ function RenderGameProgress(int $numAchievements, int $numEarnedCasual, int $num
     echo "</div>";
     echo "</div>";
 }
+
+/**
+ * Render completion icon, given that player got 100% set progress.
+ * @param   $awardedCount   How many cheevos player has gotten in set
+ * @param   $totalCount     How many cheevos set has in total
+ * @param   $hardcoreRatio  Percentage of cheevos earned on hardcore
+ * @param   $tooltip        Whether to show hover tooltip or not
+ * @return  string  The resulting HTML code
+ */
+function renderCompletionIcon(
+    int $awardedCount,
+    int $totalCount,
+    float|string $hardcoreRatio,
+    bool $tooltip = false
+): string {
+    if ($awardedCount < $totalCount) {
+        return "<div class='completion-icon'></div>";
+    }
+    $icon = $hardcoreRatio == 100.0 ? '👑' : '🎖️';
+    $class = 'completion-icon active';
+    if ($tooltip) {
+        $tooltip = $hardcoreRatio == 100.0 ? 'Mastered (hardcore)' : 'Completed';
```

Comment:
This is repurposing the `$tooltip` variable. Suggest an alternate variable name (i.e. `$tooltipText`)

Reply:
I've seen this (bad) practice somewhere else in the code. Good to know is isn't recommended, then.